tests/unittests/CoreGraphics.drawing/CGContextDrawing_DrawingOptionsTests.cpp

```diff
@@ -0,0 +1,169 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include "DrawingTest.h"
+
+static CGLineCap lineCaps[] = { kCGLineCapButt, kCGLineCapSquare, kCGLineCapRound };
+static CGLineJoin lineJoins[] = { kCGLineJoinBevel, kCGLineJoinRound };
+static CGFloat miterLimits[] = { 1.5, 1.8, 2, 3 };
+struct PhaseDashCount {
+    CGFloat phase;
+    const CGFloat dashPattern[4];
+    size_t count;
+};
+
+struct PhaseDashCount dashParams[] = { { 0, { 5, 2 }, 2 },
```

Comment:
Well the point was to test specific scenarios. Setting the phase to the first value of the dash pattern should create a line that starts with blank space. In the case of the 1, 1 dash pattern, the different phases create an inverted image. I don't see much value in creating virtually random combinations of phases and patterns. It would create more test cases, but the same can be accomplished more effectively by creating actual cases that we have an expectation for how it's meant to look. I'm not in favor of just mashing the numbers together and seeing what comes out. This is why I created the phase/dash/count struct to test particular scenarios.